accounts/src/inmemory_account_store.rs

```diff
@@ -1,21 +1,17 @@
-use std::{
-    collections::{BTreeSet, HashSet},
-    sync::Arc,
-};
+use std::{collections::HashSet, sync::Arc};
 
+use crate::account_store_interface::AccountStorageInterface;
 use async_trait::async_trait;
-use dashmap::DashMap;
+use dashmap::{DashMap, DashSet};
 use itertools::Itertools;
 use solana_lite_rpc_core::{commitment_utils::Commitment, structures::account_data::AccountData};
 use solana_rpc_client_api::filter::RpcFilterType;
 use solana_sdk::{pubkey::Pubkey, slot_history::Slot};
 use std::collections::BTreeMap;
-use tokio::sync::RwLock;
-
-use crate::account_store_interface::AccountStorageInterface;
 
 #[derive(Clone, Default)]
 pub struct AccountDataByCommitment {
+    pub pk: Pubkey,
     pub processed_accounts: BTreeMap<Slot, AccountData>,
     pub confirmed_account: Option<AccountData>,
```

Comment:
from what I see, all AccountData exist in the BTreeMap keyed by slot.
you could use `Optional<Slot>` for `confirmed_account` and `finalized_account` instead of AccountData

Reply:
Simpler to copy, anycase underlying data is stored in Arc so account data will not be really copied.